src/go/rpk/pkg/cli/cmd/cloud/namespaces_test.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2021 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package cloud_test
+
+import (
+	"bytes"
+	"errors"
+	"strings"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/cli/cmd/cloud"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/vcloud/yak"
+)
+
+func TestNamespaces(t *testing.T) {
+	tests := []struct {
+		name           string
+		client         yak.CloudApiClient
+		expectedOutput []string
+		expectedError  string
+	}{
+		{
+			name:           "success",
+			client:         &mockYakClient{},
+			expectedOutput: []string{"test", "2"},
+			expectedError:  "",
+		},
+		{
+			name:           "not logged in",
+			client:         &erroredYakClient{yak.ErrLoginTokenMissing{errors.New("inner")}},
+			expectedOutput: []string{"rpk cloud login"},
+			expectedError:  "retrieving login token",
+		},
+		{
+			name:           "generic client error",
+			client:         &erroredYakClient{errors.New("other error")},
+			expectedOutput: []string{},
+			expectedError:  "other error",
+		},
+	}
+
+	for _, tt := range tests {
+		var buf bytes.Buffer
+		logrus.SetOutput(&buf)
+		err := cloud.GetNamespaces(tt.client, &buf)
+		if len(tt.expectedOutput) > 0 {
+			for _, s := range tt.expectedOutput {
+				if !strings.Contains(buf.String(), s) {
+					t.Errorf("%s: expecting string %s in output %s", tt.name, s, buf.String())
+				}
+			}
+		}
+		if tt.expectedError != "" {
+			if err == nil || !strings.Contains(err.Error(), tt.expectedError) {
+				t.Errorf("%s: expecting error %s got %v", tt.name, tt.expectedError, err)
+			}
+		}
+	}
+}
+
+var _ yak.CloudApiClient = &mockYakClient{}
```

Comment:
Since the test struct's `client` field (L26) is of type `yak.CloudApiClient`, this is not needed.

Reply:
true, I tend to include this for every type implementing an interface though to make sure that whatever the usage changes, I always get clear compile time error when my type does not conform to interface. Do you feel strongly about removing this?

Reply:
I removed it. I still think though it does not harm and produce a better compile time errors :)